app/controllers/courses_controller.rb

```diff
@@ -155,7 +168,7 @@ def save_unlocks
   private
 
   def course_params
-    params.permit(course: [:name, :source_url, :git_branch])
+    params.require(:course).permit(:name, :title, :description, :material_url, :source_url, :git_branch)
```

Comment:
ping^
this has more consequences than one might think (e.g. how files are stored by clients)